src/sema/assembly/builtin.rs

```diff
@@ -194,6 +194,13 @@ impl AssemblyBuiltInFunction {
     }
 }
 
+impl ToString for AssemblyBuiltInFunction {
+    fn to_string(&self) -> String {
+        let prototype = self.get_prototype_info();
+        prototype.name.parse().unwrap()
+    }
+}
+
```

Comment:
Why are you using `.parse().unwrap()` rather than `.to_owned()`?
Why implement `ToString` rather than `Display`? The former does a copy/malloc.

Reply:
To maintain coherence with other constructs in Solang, I implemented the `to_string` functions. It is also implemented for `Type` and `Loc`.
Returning a String as in `fn to_string(&self) -> String` doesn't do a malloc. The compiler either moves the result or builds the value in place at the line of calling.
What advantages do you see for `Display` over `ToString`?